lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -190,10 +242,116 @@ void SetMessageManagerHandler::initiate_release_session(
    * print the fatal error and move on
    */
   if (update_success) {
-    MLOG(MINFO)
+    MLOG(MDEBUG)
         << "Successfully released and updated SessionStore of subscriber"
         << imsi;
   }
 }
 
+/* This function called when any specific  session moved to
+ * idle state
+ */
+void SetMessageManagerHandler::pdu_session_inactive(
+    const SetSmNotificationContext& notif,
+    std::function<void(Status, SmContextVoid)> response_callback) {
+  // extract values from proto
+  auto pdu_id = notif.rat_specific_notification().pdu_session_id();

```

Comment:
are these strings? let's explicitly type

Reply:
pdu_id is of type unit32_t, I have replaced auto keyword with unit32_t explicitly as suggested.